src/QsCompiler/CompilationManager/EditorSupport/CodeActions.cs

```diff
@@ -76,6 +78,42 @@ private static WorkspaceEdit GetWorkspaceEdit(this FileContentManager file, para
                 : ImmutableArray<NonNullable<string>>.Empty;
         }
 
+        /// <summary>
+        /// Returns all types that match an alternative capitalization of this type.
+        /// </summary>
+        private static IEnumerable<string> TypesWithDifferentCapitalization(CompilationUnit compilation, string type) =>
+            compilation.GetTypes().Keys
+                .Select(t => t.Name.Value)
+                .Where(t => t != type && t.Equals(type, StringComparison.OrdinalIgnoreCase));
```

Comment:
I think there are a couple issues with this:

1. It uses `compilation.GetTypes()` which only includes types defined in the current compilation unit, not types in references.
2. It looks for any matching type names based on the second part of the qualified name, but it doesn't check if the namespace of that type is opened.

Reply:
> It uses `compilation.GetTypes()` which only includes types defined in the current compilation unit, not types in references.

On second thought, this might be fine, since there is another code action that offers to open namespaces for you (but does a case-sensitive search). So in two steps you can go from an un-opened, mis-cased name to an open and correctly cased name.